rest/server_context.go

```diff
@@ -1290,7 +1295,7 @@ func CheckRoles(httpClient *http.Client, managementEndpoints []string, username,
 
 	for _, roleResult := range whoAmIResults.Roles {
 		for _, requireRole := range requestedRoles {
-			if roleResult.BucketName == bucketName && roleResult.RoleName == requireRole {
+			if (roleResult.BucketName == bucketName || roleResult.BucketName == "") && roleResult.RoleName == requireRole {
```

Comment:
Is this safe? Does an empty roleResult.BucketName mean it's a cluster-wide role?

Reply:
Yes. A missing bucket name on the result is a cluster-wide role. 
This is required because for example, on '_online' this requires the cluster scoped 'sgw_dev_ops' role. A whoami request will return this with an empty BucketName as this is cluster scoped. However, the bucketName we're checking against will not be empty as the endpoint is bucket scoped. 
I'm fairly sure this should be safe, is there a scenario you can see where it isn't safe?

Reply:
I was concerned about CheckRoles requesting a role for a specific bucket but then we're actually ignoring it in here. But if that's handled because of cluster-wide roles I think that's OK.

Reply:
Yeah I think its fine.

Reply:
As discussed we do need to allow non-db scoped roles to access db scoped endpoints.
Have made changes by adding a 'Role' type which contains information on whether the role in question is scoped.